core/unit_test/TestMultiGPU.hpp

```diff
@@ -182,14 +182,52 @@ void test_scratch(TEST_EXECSPACE exec0, TEST_EXECSPACE exec1) {
   ASSERT_EQ(error1, 0);
 }
 
+// Test that stream synchronization behavoir for various GPU APIs matches the
+// assumptions made in Kokkos for multi gpu support, namely, that any stream (no
+// matter which device it is created on) can be synced from any device.
+struct AccumulateTag {};
+struct CheckTag {};
+
 template <int N>
-__global__ void accumulate_kernel(int *value) {
-  for (int i = 0; i < N; ++i) {
-    Kokkos::atomic_inc(value);
+struct StreamSyncFunctor {
+  int *value;
+  int *check;
+
+  StreamSyncFunctor(int *value_, int *check_) : value(value_), check(check_) {}
```

Comment:
I think you could write this as an aggregate https://en.cppreference.com/w/cpp/language/aggregate_initialization